help/en/html/tools/logixng/reference/chapter5.shtml

```diff
@@ -1789,6 +1842,45 @@ LogixNG: IQ:AUTO:0001
 
       <!-- = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = -->
 
+      <h4 id="execute-program">Execute program<span class="since">since 5.13.6</span></h4>
+
+      <p><strong>Execute program</strong> is used to execute a program/script outside of JMRI.  The
+      program has no access to the JMRI run-time environment.  The
+      <a href="chapter6.shtml#fileasflag">File as flag</a> expression can be used to notify JMRI of
+      the program state.  Program output is not captured.  This includes print and echo statements.
+      Using <em>redirect</em> does not work. To retain output, it must be written to a file.</p>
+
+      <p>The dialog has 3 tabs. The <strong>Program</strong> tab has the name of the program and if
+      necessary, the path to the program.  This will vary depending on the operating system. In the
+      image it is a script in the <em>scripts</em> directory within the <em>user files location</em>
+      using the macOS/Linux command line format.</p>
+
+      <div style="margin-left: 2em">
+        <a href="images/chapter5/execute_program.png">
+        <img src="images/chapter5/execute_program.png"
+        alt="Chapter 5 execute program action"></a>
+      </div>
+
+      <p>The <strong>Parameters</strong> tab is used to pass command line parameters to the program.</p>
```

Comment:
Each parameter must be given separately. For example, the command `ls -al *.java` must be given as command `ls` with the parameters `-al` and `*.java`. So if the user tries to execute `ls` with the parameter `-al *.java`, it will not work.

For background see:
https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/lang/Runtime.html#exec(java.lang.String)
https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/lang/Runtime.html#exec(java.lang.String%5B%5D)